app/models/manageiq/providers/kubevirt/builder.rb

```diff
@@ -30,12 +30,19 @@ def build_inventory(manager, target)
         collector = ManageIQ::Providers::Kubevirt::Inventory::Collector.new(manager, target)
         collector.nodes = {}
         manager.with_provider_connection do |connection|
-          collector.offline_vms = connection.offline_vm(name)
-          collector.live_vms = connection.live_vm(name)
-          collector.templates = connection.template(name)
+          if target.template?
+            collector.templates = [connection.template(name)]
+          else
+            collector.offline_vms = [connection.offline_vm(name)]
+            begin
+              collector.live_vms = [connection.live_vm(name)]
+            rescue
+              # target refresh of an offline vm might fail if it has no live vm
```

Comment:
should we log here something here. Maybe in debug.

Reply:
added log